napari/utils/io.py

```diff
@@ -306,12 +348,121 @@ def read_csv(filename: str) -> Tuple[np.array, List[str]]:
     column_names : list
         List of column names for table data.
     """
-    with open(filename, newline='') as csvfile:
-        reader = csv.reader(csvfile, delimiter=',')
-        output_data = []
-        for row_index, row in enumerate(reader):
-            if row_index == 0:
-                column_names = [str(i) for i in row]
-            else:
-                output_data.append([float(i) for i in row])
-        return np.array(output_data), column_names
+    column_names, *data = iter_csv(filename)
+    return np.array(data), column_names
```

Comment:
I kinda feel like `read_csv` should be what `csv_to_layer_data` is now...

Reply:
We can drop `read_csv` and rename `csv_to_layer_data` as `read_csv_layer_data` might be a bit more keeping with our other names

Reply:
actually keeping `read_csv` for now is nice, but i have renamed `csv_to_layer_data` to `read_csv_layer_data`